app/components/forms/wizard/custom-form-rich-text.ts

```diff
@@ -0,0 +1,248 @@
+  field: CustomForm | null,
+  customForms: CustomForm[],
+  form: string,
+  event: any,
+  formIdentifier: string | '',
+  mainLanguage: string | 'en',
+  min: number | 0,
+  max: number | 10,
+  onSave: (() => void) | null
+}
+
+export default class CustomFormInput extends Component<Args> {
+  @tracked
+  name = '';
+
+  @tracked
+  type = 'text';
+
+  @service
+  store!: DS.Store;
+
+  @tracked
+  min = 0;
+
+  @tracked
+  max = 10;
+
+  @tracked
+  mainLanguage = 'en'
+
+  @tracked
+  selectedLanguage:string[] = [this.mainLanguage]
+
+  @tracked
+  subForm:SubForm[] = A([]);
+
+  translations:Translate[] = [];
+
+  @action
+  updated(): void {
+    this.subForm.clear()
+    if (this.args.field) {
+      this.name = this.args.field.name;
+      this.type = this.args.field.type;
+      this.min = this.args.field.min;
+      this.max = this.args.field.max;
+      this.mainLanguage = this.args.field.mainLanguage;
+      this.translations = this.args.field.translations;
+      this.selectedLanguage.clear();
+      this.selectedLanguage.pushObject(this.mainLanguage)
+      const selectedLanguage = this.translations?.map(trans => trans.code || trans.language_code);
+      this.selectedLanguage.pushObjects(selectedLanguage)
+      this.translations?.forEach((trans: Translate) => {
+        const { name, code, language_code, form_id, isDeleted, id } = trans;
+        if (isDeleted || (name && (code || language_code))) {
+          this.subForm.pushObject({
+            id,
+            form_id,
+            name,
+            languages       : translateLanguages,
+            ignoreLanguages : this.selectedLanguage,
+            selectedLang    : code || language_code,
+            isDeleted
+          })
+        }
+      })
+    } else {
+      this.name = '';
+      this.min = 0;
+      this.max = 10;
+      this.mainLanguage = 'en';
+    }
+  }
+
+  @computed('name')
+  get identifier(): string {
+    return getIdentifier(this.name, this.args.customForms);
+  }
+
+  @computed('name', 'selectedLanguage.@each', 'subForm.@each.name')
+  get validIdentifier(): boolean {
+    const nameValid = this.identifier.trim().length > 0 && this.name.trim().length > 0;
+    const transInValid = this.subForm.filter(field => !field.isDeleted && (!field.name || !field.selectedLang))
+    return nameValid && !transInValid?.length;
+  }
+
+  get languageList(): object[] {
+    return translateLanguages.filter(language =>
+      language.code === this.mainLanguage || !this.selectedLanguage.includes(language.code)
+    )
+  }
+
+  // @computed('translations')
+  get translationsList(): Translate[] {
+    const translations: Translate[] = []
+    this.subForm.forEach(field => {
+      const { id, form_id, name, isDeleted, selectedLang } = field
+      if (isDeleted || (name && selectedLang)) {
+        translations.pushObject({
+          id,
+          form_id,
+          name,
+          code          : selectedLang,
+          language_code : selectedLang,
```

Comment:
what's the difference between code and language_code ?

Reply:
current locale and selected language
@lthanhhieu perhaps a better naming convention